subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportImplementation.kt

```diff
@@ -335,11 +333,29 @@ class ReportImplementation(
         }
     }
 
-    private inline fun <reified T : ReportState> getCastedState(
-        exceptionBuilder: (String) -> IllegalStateException = { IllegalStateException(it) }
-    ): T {
-        return getCastedStateOrNull()
-            ?: throw exceptionBuilder("Invalid state. Expected ${T::class.java} actual $state")
+    private inline fun <reified T : ReportState> getCastedState(cause: Throwable? = null): T {
+        val castedClass = getCastedStateOrNull<T>()
+
+        if (castedClass == null) {
+            val _state = state
+
+            val detailedCause = cause ?: when (_state) {
+                is ReportState.Initialized.NotStarted ->
+                    IllegalStateException("Test not started, incident = ${_state.incident}")
+
+                is ReportState.Initialized.Started -> null
```

Comment:
I disagree that we need different reactions when the expected state different from the actual.
Could you explain your motivation?

Reply:
Motivation is in PR descriptions, we reported doubles, because error message contained too much: object hashes and full toString content

i changed it to exception with `class.simpleName`s and additional `cause` which could be available or not, depending of state

Reply:
I understand the part about `state::class.java.simpleName`
But why do you use  `_state.incident` only with `ReportState.Initialized.NotStarted`?
All `ReportState.Initialized` may have `_state.incident`

Reply:
ah, that was my mistake, let me fix it real quick